x/stake/endblock.go

```diff
@@ -50,12 +56,16 @@ func EndBreatheBlock(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.
 			if !sdk.IsUpgrade(sdk.BEP128) {
 				k.Distribute(sideChainCtx, sideChainIds[i])
 			} else {
-				k.DistributeInBreathBlock(sideChainCtx, sideChainIds[i])
+				fee := k.DistributeSideChainInBreathBlock(sideChainCtx, sideChainIds[i])
```

Comment:
To me,  do account `send` is safer than do balance `sub` and `add`, because when we do `send`, the `sub` and `add` happen atomically.  So I think return the `fee` here and then do the balance `add` in `DistributeBeaconChainInBreathBlock` is the best practice, try to store the funds in a `system account` when doing `DistributeSideChainInBreathBlock`, and then transfer funds from `system account` to validators in `DistributeBeaconChainInBreathBlock`.

Reply:
Since BEP128 delayed the distribution of delegators, just calculate the rewards and change the accounts' balance in later blocks. `DistributeSideChainInBreathBlock ` and `DistributeBeaconChainInBreathBlock` are in the same `EndBlock` function, they are atomic naturally. We can only change the calculation, affect the rewards data saved, thus reducing the modification of bank storage.

Reply:
changed the logics